homeassistant/components/device_tracker/google_maps.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for Google Maps location sharing.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.google_maps/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_GPS)
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['locationsharinglib==0.3.0']
+
+CREDENTIALS_FILE = 'google_maps_location_sharing.conf'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_scanner(hass, config: ConfigType, see, discovery_info=None):
+    """Set up the scanner."""
+    scanner = GoogleMapsScanner(hass, config, see)
+    return scanner if scanner.success_init else None
+
+
+class GoogleMapsScanner(object):
+    """Representation of an Google Maps location sharing account."""
+
+    def __init__(self, hass, config: ConfigType, see) -> None:
+        """Initialize the scanner."""
+        from locationsharinglib import Service
+        from locationsharinglib.locationsharinglibexceptions import InvalidUser
+
+        self.hass = hass
```

Comment:
Nothing is using the instance attribute outside this method. Please remove the attribute and just use `hass` locally in this method.